src/module.c

```diff
@@ -1500,7 +1500,7 @@ initpycurl(void)
     Py_DECREF(arglist);
     PyDict_SetItemString(d, "CurlSockAddr", curl_sockaddr_type);
 
-#if defined(WITH_THREAD) && (PY_MINOR_VERSION < 3 || PY_MINOR_VERSION == 3 && PY_MINOR_VERSION < 9)
+#if defined(WITH_THREAD) && (PY_MAJOR_VERSION < 3 || PY_MAJOR_VERSION == 3 && PY_MINOR_VERSION < 9)
```

Comment:
AFAICS `PyEval_InitThreads()` is a NO-OP on Python >= 3.9. 
So those `#if defined` can be dropped with a proper comment.

Reply:
Yeah, that's the whole point of this `#if` block: to avoid calling `PyEval_InitThreads()` on Python 3.9+.

Reply:
`PyEval_InitThreads()` is not dropped. It's just deprecated. The warning can easily be surpressed.
Check the CPython [repo](https://github.com/python/cpython/blob/main/Python/ceval.c#L336). Still there in 3.11.

Reply:
It is going to be removed eventually, at which point this will have to be done anyway.  End users should not have to suppress warnings themselves.